### ISSUE TITLE
Blank Avatar

### ISSUE THREAD

**Issue by Zalaxx** [xwiki-labs/cryptpad#400]
When I try to upload my avatar, only appear a white blank square instead of the default picture. Cryptpad said that the picture size is 1.55Kb. But I have tried with .png picture (676Kb) and .jpg (153Kb). None of these works.

**Comment by Zalaxx** [comment]
So I continued my research. I still can't upload an avatar from my pc (Ubuntu 19.04, Firefox Developper Edition 70.0b1, all extensions and blocking of integrated firefox disabled (I didn't touch the about:config I had modified, the problem must come from there)). Then I tried with my phone (Android EMUI 9.1 Honor 7X, Brave 1.2.0). The image is not uploaded if the default shield is enabled, but uploaded if disabled. So I downloaded Brave on my pc (version 0.68.132) and I got the same result, the avatar only uploads if we disable the integrated shield. I ended up making a vm of Ubuntu 19.04 and running Firefox without changing any settings, and the avatar uploaded without any problem. 

In the end, no problem with Firefox as I thought it would be, but with Brave's content blocking!

**Comment by ansuz** [comment]
I just checked whether this was still the case and found that it is. Brave incorrectly flags interaction with `https://sandbox.cryptpad.info/` as a `cross-site device recognition`. It's ironic because we use a two-domain setup for a sandbox which is intended to better protect users' privacy.

As you said, upload works with the _shield_ disabled. I reported the issue to brave, and I'll try to find some time to work around their block soon. Unfortunately, nothing is logged to the console when the action is blocked, so it's difficult to identify exactly what feature is triggering their system.
